### DISCUSSION TITLE
Sync Recordings to Cloud Storage? Ways to Export Recordings Fully Assembled (Instead of 10 Sec Segments)

### DISCUSSION THREAD

**Discussion by DC651F82053AD314** [blakeblackshear/frigate#9013] [Ask A Question] [answered]
I'm sure this must have been mentioned before in multiple ways, but I was having trouble finding a specified answer.  I see talk about using API calls and writing scripts to achieve this, but nothing documented. ..

One of my Frigate deployments is 3 hours away.  I built a weatherproof box with a mini-computer and a mobile 4G connection.  I use a coral device and it's set to only record when it see's cars.

I'd like to upload the recordings folder to my cloud storage via rclone sync (always keeping a 1 to 1 copy of every video recording in Frigate also available in the cloud), but it would be awesome if there were a way for the recordings to be fully assembled like they are in the Frigate interface, but in the cloud.

Now, I know that Frigate breaks recordings into 10 second segments for good reason, and looking at the raw files in the recordings path isn't the way it's meant to be used -- so, I wondered if there was some easy/documented way to have recordings saved in a separate directory, but fully assembled instead of in segments.  Then I could use rclone to sync those complete recording files to my cloud and have the option of viewing/streaming the recordings directly from there.

I'm very happy connecting to Frigate's browser GUI remotely over my VPN, however, I'd like the full recordings to sync to my cloud too incase anything happens to the remote system running the Frigate deployment.

Thanks for the excellent software and thriving community!  Sorry if this has been answered already, please point me in the right direction, if so.

**Comment by NickM-27** [comment, marked as answer]
I don't know what you mean by not documented. The entire frigate api is documented including multiple APIs that can be used for this. 

There is also a new exports api in 0.13 which is exactly what you're asking for https://deploy-preview-6262--frigate-docs.netlify.app/integrations/api#post-apiexportcamerastartstart-timestampendend-timestamp

**Reply by DC651F82053AD314** [reply]
Thanks Nick!  I just needed a point in the right direction.  I did a search and was reading a lot of older discussion about it from some years back.  I'll dig into the exports API documentation you provided.

**Comment by tboyko** [comment]
I had a similar use case and created a docker container to solve it. You might find it useful, too. [tboyko/frigate-echo](https://github.com/tboyko/frigate-echo)
